server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServerResourceManager.java

```diff
@@ -253,8 +254,10 @@ public TabletServerResourceManager(ServerContext context) {
     final AccumuloConfiguration acuConf = context.getConfiguration();
 
     long maxMemory = acuConf.getAsBytes(Property.TSERV_MAXMEM);
-    boolean usingNativeMap =
-        acuConf.getBoolean(Property.TSERV_NATIVEMAP_ENABLED) && NativeMap.isLoaded();
+    boolean usingNativeMap = acuConf.getBoolean(Property.TSERV_NATIVEMAP_ENABLED);
+    if (usingNativeMap) {
+      NativeMapLoader.load();
```

Comment:
Won't this cause the native map to get loaded twice because of the static code block being executed when `NativeMap` class is loaded? If we have this do we still need the static code block?

Reply:
No, because I baked in a `load only once` mechanism with an `AtomicBoolean`. My concern with removing this here would be that it may take some time before the `NativeMap` class is referenced, and therefore it might not exit right away. We currently have fast failure, because `NativeMap` is referenced here in the code I changed. So, when the tserver is starting up and creates an instance of this resource manager object, it will fail right away if the native maps can't be loaded. Keeping this here preserves that same fast-failure behavior.

Reply:
You probably didn't think any changes needed to be made but I still think it is presumptuous to merge a PR and mark comments resolved without an approval of the PR. When I open a PR and if I don't change the code that someone commented on, I usually let them decide if the conversation is resolved or not. I know we don't have explicit rules for these type of situations, I just find that it is the courteous thing to do when someone takes the time to review my code.

Reply:
I wasn't trying to be discourteous. I was just trying to be expeditious. I apologize if I upset you by pushing it through too quickly.

Regarding marking this thread as resolved. I thought I answered your questions. I had no reason to believe there will be follow-on and don't feel it necessary to hold open threads indefinitely when a reply is not expected and may never come. Open threads clutter the page and make it harder for me to find open comments that I haven't addressed. Why not just mark the thread as "Unresolved" if there's more to address? That's what I do. After all, I'm not intentionally resolving threads to be discourteous... I'm doing so to track outstanding tasks. Resolving the thread merely indicates that *I* don't think there's more to do there. I won't be offended if you simply reopen the thread because *you* believe there's more.

As always, please feel free to directly email me or message me on Slack or by any other means, if you feel I've done something discourteous. I'm very busy and don't always immediately see the comments on GitHub, so contacting me directly to bring it to my attention allows me to be more expeditious in correcting any misunderstandings, or apologizing when needed.

As for your observation about the static initializer. I thought I answered your question already. I understood you to mean it was redundant with the call to load() in TabletServerResourceManager, but as I answered already, I left it in because I believed it preserved existing behavior, if the NativeMap class were referenced by some other code path, either now, or in the future. As it turns out, that was a mistake, and I should have removed it, as I've done in #2453.